src/rtsp/setup.rs

```diff
@@ -0,0 +1,186 @@
+    match req {
+        SetupRequest::SenderInfo(info) => {
+            let (event_listener, event_port) = match open_tcp(local_addr).await {
+                Ok(res) => res,
+                Err(err) => {
+                    tracing::error!(%err, "failed to open event channel");
+                    return Err(StatusCode::INTERNAL_SERVER_ERROR);
+                }
+            };
+            let (event_task, event_handle) = abortable(event_handler(event_listener));
+
+            let sender = SenderHandle { info, event_handle };
+            // TODO : timing?
+            tokio::spawn(event_task);
+            tracing::info!(%event_port, "events handler spawned");
+
+            // TODO : deal with timing
+            let response = BinaryPlist(SetupResponse::Initial {
+                event_port,
+                timing_port: 0,
+                timing_peer_info: Some(TimingPeerInfo {
+                    id: adv_data.mac_addr.to_string(),
+                    addresses: vec![local_addr],
+                }),
+            });
+            *state.sender.write().unwrap() = Some(sender);
+
+            Ok(response)
+        }
+
+        SetupRequest::Streams { streams } => {
+            let mut streams_out = Vec::with_capacity(streams.len());
+            for info in streams {
+                // TODO : may be tcp or udp
+                let (data_socket, local_data_port) = match open_tcp(local_addr /*, None*/).await {
+                    Ok(res) => res,
+                    Err(err) => {
+                        tracing::error!(%err, "failed to open data channel");
+                        return Err(StatusCode::INTERNAL_SERVER_ERROR);
+                    }
+                };
+                let (data_task, data_handle) = abortable(data_handler(data_socket));
+                let (control_socket, local_control_port) = match open_udp(
+                    local_addr,
+                    info.remote_control_port
+                        .map(|port| SocketAddr::new(remote_addr.ip(), port)),
+                )
+                .await
+                {
+                    Ok(res) => res,
+                    Err(err) => {
+                        tracing::error!(%err, "failed to open control channel");
+                        return Err(StatusCode::INTERNAL_SERVER_ERROR);
+                    }
+                };
+                let (control_task, control_handle) = abortable(control_handler(control_socket));
+
+                let descriptor = StreamDescriptor {
+                    id: rand::random(),
+
+                    ty: info.ty,
+                    metadata: info.metadata.clone(),
+
+                    local_control_port,
+                    local_data_port,
+                };
+                let handle = StreamHandle {
+                    info,
+                    descriptor,
+                    data_handle,
+                    control_handle,
+                };
+                tokio::spawn(data_task);
```

Comment:
thread spawning caused by latency (we need it the less one, because of audio delay)